# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [10]:
library(tidyverse)

# Assignment 1

First, we generate some random data:

In [11]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2         V3        V4        V5         V6        V7       
1 0.8813171 0.68649134 0.5098811 0.6607987 0.88886078 0.8685517 0.9061079
2 0.8310283 0.09047892 0.3493076 0.4615235 0.09707875 0.2497208 0.4059893
  V8        V9        V10        V11       V12       V13       V14      
1 0.4943085 0.7173990 0.05797245 0.8462417 0.4220169 0.4685267 0.5900392
2 0.6468448 0.3471733 0.93880939 0.1843740 0.2187187 0.7809343 0.3709197
  V15       V16       V17       V18       V19        V20      
1 0.3994213 0.6518405 0.9494964 0.1013801 0.65407178 0.9483174
2 0.4083002 0.9498351 0.8407378 0.4257306 0.01584523 0.3306875

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [12]:
df2 <- df1 %>%
gather(Key, Value, V1:V20)
  

head(df2)

Key Value     
1 V1  0.88131714
2 V1  0.83102834
3 V2  0.68649134
4 V2  0.09047892
5 V3  0.50988111
6 V3  0.34930757

## Assignment 2

We will generate some data:

In [13]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [14]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [15]:
df5 <- full_join(df3,df4, by = "id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook